ios/RNTrackPlayer/RNTrackPlayerAudioPlayer.swift

```diff
@@ -79,7 +79,67 @@ public class RNTrackPlayerAudioPlayer: QueuedAudioPlayer {
 
 		self.reactEventEmitter.sendEvent(withName: "playback-state", body: ["state": state.rawValue])
     }
-    
+
+    override func AVWrapper(didReceiveMetadata metadata: [AVMetadataItem]) {
+        func getMetadataItem(forIdentifier:AVMetadataIdentifier) -> String? {
+            let data: String = AVMetadataItem.metadataItems(from: metadata, filteredByIdentifier: forIdentifier).first?.stringValue ?? ""
+            return data.isEmpty ? nil : data
+        }
+
+        super.AVWrapper(didReceiveMetadata: metadata)
+        var data : Dictionary<String, Optional<String>> = [
+            "source": nil,
+            "title": nil,
+            "url": nil,
+            "artist": nil,
+            "album": nil,
+            "date": nil,
+            "genre": nil,
+        ]
+        var source: String? {
+            switch metadata.first?.keySpace {
+            case AVMetadataKeySpace.id3:
+                return "id3"
+            case AVMetadataKeySpace.icy:
+                return "icy-headers"
+            case AVMetadataKeySpace.quickTimeMetadata:
+                return "quicktime"
+            case AVMetadataKeySpace.common:
+                return "unknown"
+            default: return "unknown"
+            }
+        }
+
+        data["title"] = getMetadataItem(forIdentifier: .commonIdentifierTitle)
+        data["artist"] = getMetadataItem(forIdentifier: .commonIdentifierArtist)
+        data["album"] = getMetadataItem(forIdentifier: .commonIdentifierAlbumName)
+        data["date"] = getMetadataItem(forIdentifier: .commonIdentifierCreationDate)
+
```

Comment:
Since this PR is going into dev, should we mirror the Android implementation as far as splitting the title into artist and title? As discussed here: https://github.com/react-native-kit/react-native-track-player/issues/933#issuecomment-627222509

Improve this in v2?

Reply:
Sure, I will take care of it.

Reply:
Oh and I was planning to do a PR to the SwiftAudio repo. Hopefully it gets merged in, since the SwiftAudio addition is tiny. But a good idea indeed to have a fork, to keep track of what we are adding in for RNTP.

Reply:
Since on iOS we do not receive information that the metadata is sure to be ICY, I chose to duck type it by checking for "unknown" source and the combination of the presence of title metadata in combination with lack of artist metadata..